cg/cli/set/family.py

```diff
@@ -47,40 +48,87 @@ def family(
     customer_id: Optional[str],
 ):
     """Update information about a case."""
+
+    check_nothing_to_change(action, panels, priority, customer_id, data_analysis, data_delivery)
+
     status_db: Store = context.status_db
-    case_obj: models.Family = status_db.family(family_id)
-    if case_obj is None:
-        LOG.error("Can't find case %s,", family_id)
-        raise click.Abort
-    if not any([action, panels, priority, customer_id, data_analysis, data_delivery]):
-        LOG.error("Nothing to change")
-        raise click.Abort
+    case: Family = get_case(family_id, status_db)
+
     if action:
-        LOG.info("Update action: %s -> %s", case_obj.action or "NA", action)
-        case_obj.action = action
+        update_action(case, action)
+
     if customer_id:
-        customer_obj: models.Customer = status_db.customer(customer_id)
-        if customer_obj is None:
-            LOG.error("Unknown customer: %s", customer_id)
-            raise click.Abort
-        LOG.info(f"Update customer: {case_obj.customer.internal_id} -> {customer_id}")
-        case_obj.customer = customer_obj
+        update_customer(case, customer_id, status_db)
+
     if data_analysis:
-        LOG.info(f"Update data_analysis: {case_obj.data_analysis or 'NA'} -> {data_analysis}")
-        case_obj.data_analysis = data_analysis
+        update_data_analysis(case, data_analysis)
+
     if data_delivery:
-        LOG.info(f"Update data_delivery: {case_obj.data_delivery or 'NA'} -> {data_delivery}")
-        case_obj.data_delivery = data_delivery
+        update_data_delivery(case, data_delivery)
+
     if panels:
-        for panel_id in panels:
-            panel_obj: models.Panel = status_db.panel(panel_id)
-            if panel_obj is None:
-                LOG.error(f"unknown gene panel: {panel_id}")
-                raise click.Abort
-        LOG.info(f"Update panels: {', '.join(case_obj.panels)} -> {', '.join(panels)}")
-        case_obj.panels = panels
+        update_panels(case, panels, status_db)
+
     if priority:
-        LOG.info(f"Update priority: {case_obj.priority.name} -> {priority.name}")
-        case_obj.priority = priority
+        update_priority(case, priority)
 
     status_db.commit()
+
+
+def check_nothing_to_change(action, panels, priority, customer_id, data_analysis, data_delivery):
+    if not any([action, panels, priority, customer_id, data_analysis, data_delivery]):
+        LOG.error("Nothing to change")
+        raise click.Abort
+
+
+def get_case(case_id: str, status_db: Store):
+    case: Family = status_db.family(case_id)
+
+    if case is None:
+        LOG.error(f"Can't find case {case_id}")
+        raise click.Abort
+
+    return case
+
+
+def update_action(case: Family, action: str) -> None:
```

Comment:
There are these in cg/meta/orders/case_submitter.py:

```
               self._append_ticket(ticket_id=ticket_id, case=status_db_case)
                self._update_action(action=CaseActions.ANALYZE, case=status_db_case)
                self._update_case_panel(panels=case["panels"], case=status_db_case)
```
Would be good to define the functions once and in a update.py handler to keep them collected and easy to find

Reply:
Yes, I think this would be good. Maybe we could have the following directory structure for our handlers:

```
handlers/
    ├── base.py
    ├── base_find_handler.py
    ├── base_update_handler.py
    ├── base_delete_handler.py
    ├── family/
    │   ├── __init__.py
    │   ├── family_find_handler.py
    │   ├── family_update_handler.py
    │   └── family_delete_handler.py
```